clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -185,7 +185,18 @@ public String toString() {
          * Another reason is because we make use of the list ordering to optimize the preparation of
          * incremental fetch requests (see below).
          */
-        private LinkedHashMap<TopicPartition, PartitionData> next = new LinkedHashMap<>();
+        private LinkedHashMap<TopicPartition, PartitionData> next;
```

Comment:
I'd suggest we make one step further, to just make the `sessionPartitions` a local variable only: this is because in later handling logic after the response is sent back, we only need `sessionPartitions.keySet / size` only, which we could maintain separately as a member field which is populated after the build() call. WDYT?

Reply:
NVM I realized now you also need the `sessionPartitions` for saving one fetch state copying as well.